### PR TITLE
Clarify pattern of charm build and layers/interfaces

### PR DESCRIPTION
**Pull request by AdamIsrael**
This merge attempts to clarify how `charm build` uses layers/interfaces, and the questions raised by issue #853

### CODE REVIEW

**Comment by jamesbeedy** [comment]
+1

File: src/en/developer-getting-started.md [hunk 1]
```diff
@@ -95,14 +95,22 @@ makes layers possible.
...
+The `layers` directory contains the source code of the layered charm covered in
+our examples. The `interfaces` directory is where you'd place any
+[interface-layers](./charms-layers-interfaces.md) you may wish to write, and the
+`charms` directory holds the assembled, ready to deploy charm.
 
 ```bash
+mkdir $HOME/charms
 export JUJU_REPOSITORY=$HOME/charms
+
+mkdir $JUJU_REPOSITORY/{interfaces,layers}
```
Comment (marcoceppi) [inline src/en/developer-getting-started.md]: It seems it'd be better to have the following:

```
export JUJU_REPOSITORY=$HOME/charms
export LAYER_PATH=$JUJU_REPOSITORY/layers
export INTERFACE_PATH=$JUJU_REPOSITORY/interfaces

mkdir -p $JUJU_REPOSITORY $LAYER_PATH $INTERFACE_PATH
```

simpler workflow IMO

File: src/en/developer-layer-example.md [hunk 2]
```diff
@@ -20,18 +20,22 @@ available.
...
 #### LAYER_PATH
 
-Defines the location on disk to search for local layers
+Defines the location on disk to search for local layers. First, create the base
+directory, if it doesn't exist:
 
-#### INTERFACE_PATH
+```bash
+mkdir $HOME/charms
+export JUJU_REPOSITORY=$HOME/charms
```
Comment (marcoceppi) [inline src/en/developer-layer-example.md]: Again, define the variable, then mkdir, and always `-p`

File: src/en/developer-layers-interfaces.md [hunk 3]
```diff
@@ -93,6 +93,15 @@ class MyRelationClient(RelationBase):
 
 Begin by making an interface repository if you don't currently have one.
 
+Create the base charms directory, if it doesn't exist:
+
+```bash
+mkdir $HOME/charms
```
Comment (marcoceppi) [inline src/en/developer-layers-interfaces.md]: swap operations

**Comment by evilnick** [comment]
when you finish reviewing or amending per review, probably worth a @ ping in a comment!

:cookie:

**Pull request merged**